### PR TITLE
Channel loading

### PR DESCRIPTION
**Pull request by hardening**
Reworked channel loading, comments / remarks are welcome

### CODE REVIEW

File: client/common/cmdline.c [hunk 1]
```diff
@@ -3594,45 +3594,68 @@ static BOOL freerdp_client_load_static_channel_addin(rdpChannels* channels, rdpS
 	return FALSE;
 }
 
-BOOL freerdp_client_load_addins(rdpChannels* channels, rdpSettings* settings)
+typedef struct
 {
-	UINT32 index;
+	int settingId;
```
Comment (akallabeth) [inline client/common/cmdline.c:3599]: `size_t`, not `int`

File: client/common/cmdline.c [hunk 2]
```diff
@@ -3594,45 +3594,68 @@ static BOOL freerdp_client_load_static_channel_addin(rdpChannels* channels, rdpS
...
+	const char* channelName;
+	void* args;
+} ChannelToLoad;
 
-	/* Always load FreeRDP advanced input dynamic channel */
-#if defined(CHANNEL_AINPUT_CLIENT)
-	{
-		const char* p[] = { AINPUT_CHANNEL_NAME };
+#ifndef ARRAY_SIZE
```
Comment (akallabeth) [inline client/common/cmdline.c:3604]: duplicate, there is already `ARRAYSIZE` from winpr

File: client/common/cmdline.c [hunk 3]
```diff
@@ -3594,45 +3594,68 @@ static BOOL freerdp_client_load_static_channel_addin(rdpChannels* channels, rdpS
...
+		{ FreeRDP_AudioPlayback, RDPSND_CHANNEL_NAME, NULL },
+		{ FreeRDP_RedirectClipboard, CLIPRDR_SVC_CHANNEL_NAME, NULL },
+#if defined(CHANNEL_ENCOMSP_CLIENT)
+		{ FreeRDP_EncomspVirtualChannel, ENCOMSP_SVC_CHANNEL_NAME, settings },
+#endif
+		{ FreeRDP_RemdeskVirtualChannel, REMDESK_SVC_CHANNEL_NAME, settings },
+		{ FreeRDP_RDP2TCPArgs, RDP2TCP_DVC_CHANNEL_NAME, settings->RDP2TCPArgs },
+		{ FreeRDP_RemoteApplicationMode, RAIL_SVC_CHANNEL_NAME, settings }
+	};
+	UINT32 i;
```
Comment (akallabeth) [inline client/common/cmdline.c:3639]: `size_t` not `UINT32`

**Review by akallabeth — COMMENTED** [review]
Overall LGTM, except some minor remarks.

**Review by akallabeth — APPROVED** [review]
:+1:

**Review by mfleisz — APPROVED** [review]

**Pull request merged**